client/state/themes/hooks/use-add-external-managed-theme-to-cart.tsx

```diff
@@ -0,0 +1,72 @@
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { addQueryArgs } from '@wordpress/url';
+import page from 'page';
+import { useState } from 'react';
+import 'calypso/state/themes/init';
+import { useSelector } from 'react-redux';
+import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
+import { getExternalManagedThemeProduct } from '../selectors/get-external-managed-theme-product';
+
+export type AddExternalManagedThemeToCartFunction = () => void;
+export type UseAddExternalManagedThemeToCart = {
+	id: string;
+	siteSlug: string;
+	siteEligibleForManagedExternalThemes: boolean;
+	isPurchased: boolean;
+};
+
+/**
+ * Adds a External Managed theme to the cart.
+ *
+ * @param {UseAddExternalManagedThemeToCart} options Options object
+ * @returns {[ AddExternalManagedThemeToCartFunction, boolean ]}
+ */
+export function useAddExternalManagedThemeToCart( {
+	id,
+	siteSlug,
+	siteEligibleForManagedExternalThemes,
```

Comment:
Can we call this `isSiteEligibleForManagedExternalThemes` to clearly indicate that it's a boolean? We could also consider naming this `isSiteEligibleForMarketplaceThemes` to simplify the name.
```suggestion
	isSiteEligibleForManagedExternalThemes,
```